### ISSUE TITLE
loss function 

### ISSUE THREAD

**Issue by SkiddieAhn** [addb-swstarlab/OANet#10]
A(code): loss = `(1-self.lamb)*F.mse_loss(output, target) + self.lamb*dot_loss `
 B(paper): 𝐿𝑜𝑠𝑠 = 𝑀𝑆𝐸(𝑦, 𝑦̂) + 𝑆𝑂𝑅(ℎ̃x), 𝑆𝑂𝑅(𝑀) = 𝜆‖𝑀 ∙ 𝑀𝑇 − `𝐼‖^2
 
 1. Can I think that A and B are the same? If so, I wonder what role (self.lamb) plays and what is the performance change depending on (self.lamb).
 
 2. What does 𝑀 mean in the paper?

**Comment by DELAB-Y** [comment]
1.
Yes you can think A and B are the same.
Dot loss refers to the value of soft orthogonality regularization (SOR), which calculates how well h, a two-dimensional matrix, grouped or orthogonalized the knobs. (self.lamb) acts as a weight to indicate how much importance to give to dot loss when training the model. Higher values of (self.lamb) result in better grouping of knobs. However, if (self.lamb) is too high, mse_loss will be reflected less, which can lead to poor learning. For this reason, you should set lamb high enough to take dot_loss into account, but not so high that the influence of mse_loss is ignored.

2.
M is a two-dimensional matrix consisting of several vectors. The SOR is to determine whether the vectors in that matrix are orthogonal. In this paper, M can be regarded as a reshaped h.

Thank you for the comments!
